### PR TITLE
temporary Windows CI instance for debugging

### PR DESCRIPTION
**Pull request by garyverhaegen-da**
This is made as a PR for audit/review purposes.

The goal of this change is to add a single machine, configured exactly like our Windows CI machines, so that @aherrmann-da can debug Bazel 1.1 behaviour on it, as the issue does not seem to be reproducible locally.

The plan is:
- I run `terraform apply` on this PR (I have verified that Terraform does not currently find any difference between `master` and our running infra).
- I manually, through the GCP web console, set a dummy password for that machine's RDP connection and transmit that to @aherrmann-da through Slack.
- @aherrmann-da debugs the issue.
- I run `terraform apply` on master again, cleaning up this PR's changes.

There is no need for this PR to be merged.

### CODE REVIEW

File: infra/TEMP-vsts_agent_windows_DEBUG.tf [hunk 1]
```diff
@@ -0,0 +1,166 @@
+# Copyright (c) 2019 The DAML Authors. All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+locals {
+  DEBUG_vsts_token   = "${secret_resource.vsts-token.value}"
+  DEBUG_vsts_account = "digitalasset"
```
Comment (cocreature) [inline infra/TEMP-vsts_agent_windows_DEBUG.tf]: I’ll never understand where things are called `digital-asset` and where they are called `digitialasset`.

**Comment by garyverhaegen-da** [comment]
Here is the result of applying that PR:
```
$ git rev-parse HEAD
f532ae0d9563bc4836e8655d7b3c51afbcb20bf9
$ terraform apply
secret_resource.vsts-token: Refreshing state... (ID: -)
google_compute_health_check.hoogle-http: Refreshing state... (ID: hoogle-http)
data.template_file.vsts-agent-linux-startup: Refreshing state...
google_compute_health_check.hoogle-https: Refreshing state... (ID: hoogle-https)
google_compute_global_address.default: Refreshing state... (ID: daml-nix-cache-address)
google_storage_bucket.data: Refreshing state... (ID: daml-data)
google_compute_network.hoogle: Refreshing state... (ID: hoogle-network)
google_compute_global_address.hoogle: Refreshing state... (ID: hoogle)
google_storage_bucket.default: Refreshing state... (ID: daml-nix-cache)
google_compute_global_address.default: Refreshing state... (ID: daml-bazel-cache-address)
data.google_project.current: Refreshing state...
google_storage_bucket.default: Refreshing state... (ID: daml-bazel-cache)
google_compute_instance_template.vsts-agent-windows: Refreshing state... (ID: vsts-agent-windows-20190521105417788000000001)
data.google_project.current: Refreshing state...
google_service_account.writer: Refreshing state... (ID: projects/da-dev-gcp-daml-language/servi...-daml-language.iam.gserviceaccount.com)
data.google_project.current: Refreshing state...
google_compute_instance_template.vsts-agent-linux: Refreshing state... (ID: vsts-agent-linux-20190927124452202300000001)
google_compute_backend_bucket.default: Refreshing state... (ID: daml-bazel-cache-backend)
google_compute_backend_bucket.default: Refreshing state... (ID: daml-nix-cache-backend)
google_storage_bucket_object.nix-cache-info: Refreshing state... (ID: daml-nix-cache-nix-cache-info)
google_compute_firewall.hoogle: Refreshing state... (ID: hoogle-firewall)
google_compute_instance_template.hoogle: Refreshing state... (ID: hoogle-20190927132021025900000001)
google_compute_region_instance_group_manager.vsts-agent-windows: Refreshing state... (ID: da-dev-gcp-daml-language/us-east4/vsts-agent-windows)
google_compute_region_instance_group_manager.vsts-agent-linux: Refreshing state... (ID: vsts-agent-linux)
google_compute_instance_group_manager.hoogle: Refreshing state... (ID: da-dev-gcp-daml-language/us-east4-a/hoogle)
google_compute_url_map.default: Refreshing state... (ID: daml-bazel-cache)
google_compute_url_map.default: Refreshing state... (ID: daml-nix-cache)
google_storage_bucket_acl.default: Refreshing state... (ID: daml-nix-cache-acl)
google_storage_bucket_iam_member.nix_cache_writer: Refreshing state... (ID: daml-nix-cache/roles/storage.objectAdmi...-daml-language.iam.gserviceaccount.com)
google_storage_bucket_iam_member.data: Refreshing state... (ID: daml-data/roles/storage.objectAdmin/ser...-daml-language.iam.gserviceaccount.com)
google_storage_bucket_iam_member.bazel_cache_writer: Refreshing state... (ID: daml-bazel-cache/roles/storage.objectAd...-daml-language.iam.gserviceaccount.com)
google_storage_bucket_acl.default: Refreshing state... (ID: daml-bazel-cache-acl)
google_compute_backend_service.hoogle-https: Refreshing state... (ID: hoogle-https)
google_compute_backend_service.hoogle-http: Refreshing state... (ID: hoogle-http)
google_storage_bucket_acl.data: Refreshing state... (ID: daml-data-acl)
google_compute_target_http_proxy.default: Refreshing state... (ID: daml-bazel-cache-http-proxy)
google_compute_target_https_proxy.default: Refreshing state... (ID: daml-bazel-cache-https-proxy)
google_compute_target_http_proxy.default: Refreshing state... (ID: daml-nix-cache-http-proxy)
google_compute_target_https_proxy.default: Refreshing state... (ID: daml-nix-cache-https-proxy)
google_compute_url_map.hoogle-https: Refreshing state... (ID: hoogle-https)
google_compute_global_forwarding_rule.http: Refreshing state... (ID: daml-bazel-cache-http)
google_compute_url_map.hoogle-http: Refreshing state... (ID: hoogle-http)
google_compute_global_forwarding_rule.https: Refreshing state... (ID: daml-bazel-cache-https)
google_compute_global_forwarding_rule.http: Refreshing state... (ID: daml-nix-cache-http)
google_compute_global_forwarding_rule.https: Refreshing state... (ID: daml-nix-cache-https)
google_compute_target_https_proxy.hoogle-https: Refreshing state... (ID: hoogle-https)
google_compute_target_http_proxy.hoogle-http: Refreshing state... (ID: hoogle-http)
google_storage_bucket_object.index_bazel_cache: Refreshing state... (ID: daml-bazel-cache-index.html)
google_storage_bucket_object.index_nix_cache: Refreshing state... (ID: daml-nix-cache-index.html)
google_compute_global_forwarding_rule.hoogle_https: Refreshing state... (ID: hoogle-https)
google_compute_global_forwarding_rule.hoogle_http: Refreshing state... (ID: hoogle-http)

An execution plan has been generated and is shown below.
Resource actions are indicated with the following symbols:
  + create

Terraform will perform the following actions:

  + google_compute_firewall.allow-rdp
      id:                                                  <computed>
      allow.#:                                             "2"
      allow.658773778.ports.#:                             "1"
      allow.658773778.ports.0:                             "3389"
      allow.658773778.protocol:                            "tcp"
      allow.989465160.ports.#:                             "1"
      allow.989465160.ports.0:                             "3389"
      allow.989465160.protocol:                            "udp"
      creation_timestamp:                                  <computed>
      destination_ranges.#:                                <computed>
      direction:                                           <computed>
      name:                                                "allow-rdp"
      network:                                             "default"
      priority:                                            "1000"
      project:                                             <computed>
      self_link:                                           <computed>
      source_ranges.#:                                     <computed>
      target_tags.#:                                       "1"
      target_tags.1007910815:                              "debugrdp"

  + google_compute_instance_template.vsts-agent-windows-DEBUG
      id:                                                  <computed>
      can_ip_forward:                                      "false"
      disk.#:                                              "2"
      disk.0.auto_delete:                                  "true"
      disk.0.boot:                                         "false"
      disk.0.device_name:                                  <computed>
      disk.0.disk_size_gb:                                 "200"
      disk.0.disk_type:                                    "pd-ssd"
      disk.0.interface:                                    <computed>
      disk.0.mode:                                         <computed>
      disk.0.source_image:                                 "windows-cloud/windows-2016"
      disk.0.type:                                         <computed>
      disk.1.auto_delete:                                  "true"
      disk.1.boot:                                         "false"
      disk.1.device_name:                                  <computed>
      disk.1.disk_size_gb:                                 "200"
      disk.1.disk_type:                                    "pd-ssd"
      disk.1.interface:                                    <computed>
      disk.1.mode:                                         <computed>
      disk.1.source_image:                                 <computed>
      disk.1.type:                                         <computed>
      labels.%:                                            "5"
      labels.cost-allocation:                              "daml-language"
      labels.host-group:                                   "buildpipeline"
      labels.infra-owner:                                  "daml-language"
      labels.managed:                                      "true"
      labels.target:                                       "infra"
      machine_type:                                        "n1-standard-8"
      metadata.%:                                          "2"
      metadata.sysprep-specialize-script-ps1:              "Set-StrictMode -Version latest\n$ErrorActionPreference = 'Stop'\n\n# Disable Windows Defender to speed up disk access\nSet-MpPreference -DisableRealtimeMonitoring $true\n\n# Enable long paths\nSet-ItemProperty -Path 'HKLM:\\SYSTEM\\CurrentControlSet\\Control\\FileSystem' -Name LongPathsEnabled -Type DWord -Value 1\n\n# Disable UAC\nNew-ItemProperty -Path HKLM:Software\\Microsoft\\Windows\\CurrentVersion\\policies\\system -Name EnableLUA -PropertyType DWord -Value 0 -Force\n\n# Redirect logs to SumoLogic\n\ncd $env:UserProfile;\nInvoke-WebRequest https://dl.google.com/cloudagents/windows/StackdriverLogging-v1-9.exe -OutFile StackdriverLogging-v1-9.exe;\n.\\StackdriverLogging-v1-9.exe /S /D=\"C:\\Stackdriver\\Logging\\\"\n\n# Install chocolatey\niex (New-Object System.Net.WebClient).DownloadString('https://chocolatey.org/install.ps1')\n\n# Install git, bash\n& choco install git --no-progress --yes 2>&1 | %{ \"$_\" }\n& choco install windows-sdk-10.1 --no-progress --yes 2>&1 | %{ \"$_\" }\n\n# Add tools to the PATH\n$OldPath = (Get-ItemProperty -Path 'Registry::HKEY_LOCAL_MACHINE\\System\\CurrentControlSet\\Control\\Session Manager\\Environment' -Name PATH).path\n$NewPath = \"$OldPath;C:\\Program Files\\Git\\bin;C:\\Program Files (x86)\\Windows Kits\\10\\App Certification Kit\"\nSet-ItemProperty -Path 'Registry::HKEY_LOCAL_MACHINE\\System\\CurrentControlSet\\Control\\Session Manager\\Environment' -Name PATH -Value $NewPath\n\necho \"== Prepare the D:\\ drive\"\n\n$partition = @\"\nselect disk 1\nclean\nconvert gpt\ncreate partition primary\nformat fs=ntfs quick\nassign letter=\"D\"\n\"@\n$partition | Set-Content C:\\diskpart.txt\n& diskpart /s C:\\diskpart.txt 2>&1 | %{ \"$_\" }\n\n# Create a temporary and random password for the VSTS user, forget about it once this script has finished running\n$Username = \"VssAdministrator\"\n$Account = \"$env:COMPUTERNAME\\$Username\"\nAdd-Type -AssemblyName System.Web\n$Password = [System.Web.Security.Membership]::GeneratePassword(24, 0)\n\necho \"== Creating the VSTS user\"\n\n#New-LocalUser $Username -Password $SecurePassword -FullName $Username\nnet user $Username $Password /add /y\n# net localgroup administrators $Username /add\nAdd-LocalGroupMember -Group \"Administrators\" -Member $Username\n\nwinrm set winrm/config/winrs '@{MaxMemoryPerShellMB=\"2048\"}'\nwinrm set winrm/config '@{MaxTimeoutms=\"1800000\"}'\nwinrm set winrm/config/service/auth '@{Basic=\"true\"}'\nnet stop winrm\nsc.exe config winrm start=auto\nnet start winrm\n\necho \"== Installing the VSTS agent\"\n\nchoco install azure-pipelines-agent --no-progress --yes --params \"'/Token:<?> /Pool:windows-pool-debug /Url:https://dev.azure.com/digitalasset/ /LogonAccount:$Account /LogonPassword:$Password /Work:D:\\a'\"\necho OK\n"
      metadata.windows-shutdown-script-ps1:                "c://agent/config remove --unattended --auth PAT --token '<?>'"
      metadata_fingerprint:                                <computed>
      name:                                                <computed>
      name_prefix:                                         "windows-debug-"
      network_interface.#:                                 "1"
      network_interface.0.access_config.#:                 "1"
      network_interface.0.access_config.0.assigned_nat_ip: <computed>
      network_interface.0.access_config.0.nat_ip:          <computed>
      network_interface.0.access_config.0.network_tier:    <computed>
      network_interface.0.address:                         <computed>
      network_interface.0.network:                         "default"
      network_interface.0.subnetwork_project:              <computed>
      project:                                             <computed>
      region:                                              <computed>
      scheduling.#:                                        "1"
      scheduling.0.automatic_restart:                      "false"
      scheduling.0.on_host_maintenance:                    "TERMINATE"
      scheduling.0.preemptible:                            "true"
      self_link:                                           <computed>
      tags.#:                                              "1"
      tags.1007910815:                                     "debugrdp"
      tags_fingerprint:                                    <computed>

  + google_compute_region_instance_group_manager.vsts-agent-windows-TEMP_DEBUG
      id:                                                  <computed>
      base_instance_name:                                  "windebug"
      distribution_policy_zones.#:                         <computed>
      fingerprint:                                         <computed>
      instance_group:                                      <computed>
      name:                                                "windows-debug"
      project:                                             <computed>
      region:                                              "us-east4"
      self_link:                                           <computed>
      target_size:                                         "1"
      update_policy.#:                                     "1"
      update_policy.0.max_surge_fixed:                     "3"
      update_policy.0.max_unavailable_fixed:               <computed>
      update_policy.0.min_ready_sec:                       "520"
      update_policy.0.minimal_action:                      "REPLACE"
      update_policy.0.type:                                "PROACTIVE"
      version.#:                                           "1"
      version.0.instance_template:                         "${google_compute_instance_template.vsts-agent-windows-DEBUG.self_link}"
      version.0.name:                                      "windows-debug"
      wait_for_instances:                                  "false"


Plan: 3 to add, 0 to change, 0 to destroy.

Do you want to perform these actions?
  Terraform will perform the actions described above.
  Only 'yes' will be accepted to approve.

  Enter a value: yes

google_compute_firewall.allow-rdp: Creating...
  allow.#:                  "" => "2"
  allow.658773778.ports.#:  "" => "1"
  allow.658773778.ports.0:  "" => "3389"
  allow.658773778.protocol: "" => "tcp"
  allow.989465160.ports.#:  "" => "1"
  allow.989465160.ports.0:  "" => "3389"
  allow.989465160.protocol: "" => "udp"
  creation_timestamp:       "" => "<computed>"
  destination_ranges.#:     "" => "<computed>"
  direction:                "" => "<computed>"
  name:                     "" => "allow-rdp"
  network:                  "" => "default"
  priority:                 "" => "1000"
  project:                  "" => "<computed>"
  self_link:                "" => "<computed>"
  source_ranges.#:          "" => "<computed>"
  target_tags.#:            "" => "1"
  target_tags.1007910815:   "" => "debugrdp"
google_compute_instance_template.vsts-agent-windows-DEBUG: Creating...
  can_ip_forward:                                      "" => "false"
  disk.#:                                              "" => "2"
  disk.0.auto_delete:                                  "" => "true"
  disk.0.boot:                                         "" => "<computed>"
  disk.0.device_name:                                  "" => "<computed>"
  disk.0.disk_size_gb:                                 "" => "200"
  disk.0.disk_type:                                    "" => "pd-ssd"
  disk.0.interface:                                    "" => "<computed>"
  disk.0.mode:                                         "" => "<computed>"
  disk.0.source_image:                                 "" => "windows-cloud/windows-2016"
  disk.0.type:                                         "" => "<computed>"
  disk.1.auto_delete:                                  "" => "true"
  disk.1.boot:                                         "" => "<computed>"
  disk.1.device_name:                                  "" => "<computed>"
  disk.1.disk_size_gb:                                 "" => "200"
  disk.1.disk_type:                                    "" => "pd-ssd"
  disk.1.interface:                                    "" => "<computed>"
  disk.1.mode:                                         "" => "<computed>"
  disk.1.source_image:                                 "" => "<computed>"
  disk.1.type:                                         "" => "<computed>"
  labels.%:                                            "" => "5"
  labels.cost-allocation:                              "" => "daml-language"
  labels.host-group:                                   "" => "buildpipeline"
  labels.infra-owner:                                  "" => "daml-language"
  labels.managed:                                      "" => "true"
  labels.target:                                       "" => "infra"
  machine_type:                                        "" => "n1-standard-8"
  metadata.%:                                          "" => "2"
  metadata.sysprep-specialize-script-ps1:              "" => "Set-StrictMode -Version latest\n$ErrorActionPreference = 'Stop'\n\n# Disable Windows Defender to speed up disk access\nSet-MpPreference -DisableRealtimeMonitoring $true\n\n# Enable long paths\nSet-ItemProperty -Path 'HKLM:\\SYSTEM\\CurrentControlSet\\Control\\FileSystem' -Name LongPathsEnabled -Type DWord -Value 1\n\n# Disable UAC\nNew-ItemProperty -Path HKLM:Software\\Microsoft\\Windows\\CurrentVersion\\policies\\system -Name EnableLUA -PropertyType DWord -Value 0 -Force\n\n# Redirect logs to SumoLogic\n\ncd $env:UserProfile;\nInvoke-WebRequest https://dl.google.com/cloudagents/windows/StackdriverLogging-v1-9.exe -OutFile StackdriverLogging-v1-9.exe;\n.\\StackdriverLogging-v1-9.exe /S /D=\"C:\\Stackdriver\\Logging\\\"\n\n# Install chocolatey\niex (New-Object System.Net.WebClient).DownloadString('https://chocolatey.org/install.ps1')\n\n# Install git, bash\n& choco install git --no-progress --yes 2>&1 | %{ \"$_\" }\n& choco install windows-sdk-10.1 --no-progress --yes 2>&1 | %{ \"$_\" }\n\n# Add tools to the PATH\n$OldPath = (Get-ItemProperty -Path 'Registry::HKEY_LOCAL_MACHINE\\System\\CurrentControlSet\\Control\\Session Manager\\Environment' -Name PATH).path\n$NewPath = \"$OldPath;C:\\Program Files\\Git\\bin;C:\\Program Files (x86)\\Windows Kits\\10\\App Certification Kit\"\nSet-ItemProperty -Path 'Registry::HKEY_LOCAL_MACHINE\\System\\CurrentControlSet\\Control\\Session Manager\\Environment' -Name PATH -Value $NewPath\n\necho \"== Prepare the D:\\ drive\"\n\n$partition = @\"\nselect disk 1\nclean\nconvert gpt\ncreate partition primary\nformat fs=ntfs quick\nassign letter=\"D\"\n\"@\n$partition | Set-Content C:\\diskpart.txt\n& diskpart /s C:\\diskpart.txt 2>&1 | %{ \"$_\" }\n\n# Create a temporary and random password for the VSTS user, forget about it once this script has finished running\n$Username = \"VssAdministrator\"\n$Account = \"$env:COMPUTERNAME\\$Username\"\nAdd-Type -AssemblyName System.Web\n$Password = [System.Web.Security.Membership]::GeneratePassword(24, 0)\n\necho \"== Creating the VSTS user\"\n\n#New-LocalUser $Username -Password $SecurePassword -FullName $Username\nnet user $Username $Password /add /y\n# net localgroup administrators $Username /add\nAdd-LocalGroupMember -Group \"Administrators\" -Member $Username\n\nwinrm set winrm/config/winrs '@{MaxMemoryPerShellMB=\"2048\"}'\nwinrm set winrm/config '@{MaxTimeoutms=\"1800000\"}'\nwinrm set winrm/config/service/auth '@{Basic=\"true\"}'\nnet stop winrm\nsc.exe config winrm start=auto\nnet start winrm\n\necho \"== Installing the VSTS agent\"\n\nchoco install azure-pipelines-agent --no-progress --yes --params \"'/Token:<?> /Pool:windows-pool-debug /Url:https://dev.azure.com/digitalasset/ /LogonAccount:$Account /LogonPassword:$Password /Work:D:\\a'\"\necho OK\n"
  metadata.windows-shutdown-script-ps1:                "" => "c://agent/config remove --unattended --auth PAT --token '<?>'"
  metadata_fingerprint:                                "" => "<computed>"
  name:                                                "" => "<computed>"
  name_prefix:                                         "" => "windows-debug-"
  network_interface.#:                                 "" => "1"
  network_interface.0.access_config.#:                 "" => "1"
  network_interface.0.access_config.0.assigned_nat_ip: "" => "<computed>"
  network_interface.0.access_config.0.nat_ip:          "" => "<computed>"
  network_interface.0.access_config.0.network_tier:    "" => "<computed>"
  network_interface.0.address:                         "" => "<computed>"
  network_interface.0.network:                         "" => "default"
  network_interface.0.subnetwork_project:              "" => "<computed>"
  project:                                             "" => "<computed>"
  region:                                              "" => "<computed>"
  scheduling.#:                                        "" => "1"
  scheduling.0.automatic_restart:                      "" => "false"
  scheduling.0.on_host_maintenance:                    "" => "TERMINATE"
  scheduling.0.preemptible:                            "" => "true"
  self_link:                                           "" => "<computed>"
  tags.#:                                              "" => "1"
  tags.1007910815:                                     "" => "debugrdp"
  tags_fingerprint:                                    "" => "<computed>"
google_compute_instance_template.vsts-agent-windows-DEBUG: Creation complete after 6s (ID: windows-debug-20191031135035762000000001)
google_compute_region_instance_group_manager.vsts-agent-windows-TEMP_DEBUG: Creating...
  base_instance_name:                    "" => "windebug"
  distribution_policy_zones.#:           "" => "<computed>"
  fingerprint:                           "" => "<computed>"
  instance_group:                        "" => "<computed>"
  name:                                  "" => "windows-debug"
  project:                               "" => "<computed>"
  region:                                "" => "us-east4"
  self_link:                             "" => "<computed>"
  target_size:                           "" => "1"
  update_policy.#:                       "" => "1"
  update_policy.0.max_surge_fixed:       "" => "3"
  update_policy.0.max_unavailable_fixed: "" => "<computed>"
  update_policy.0.min_ready_sec:         "" => "520"
  update_policy.0.minimal_action:        "" => "REPLACE"
  update_policy.0.type:                  "" => "PROACTIVE"
  version.#:                             "" => "1"
  version.0.instance_template:           "" => "https://www.googleapis.com/compute/beta/projects/da-dev-gcp-daml-language/global/instanceTemplates/windows-debug-20191031135035762000000001"
  version.0.name:                        "" => "windows-debug"
  wait_for_instances:                    "" => "false"
google_compute_firewall.allow-rdp: Creation complete after 8s (ID: allow-rdp)
google_compute_region_instance_group_manager.vsts-agent-windows-TEMP_DEBUG: Still creating... (10s elapsed)
google_compute_region_instance_group_manager.vsts-agent-windows-TEMP_DEBUG: Creation complete after 18s (ID: da-dev-gcp-daml-language/us-east4/windows-debug)

Apply complete! Resources: 3 added, 0 changed, 0 destroyed.

Outputs:

bazel_cache_ip = 35.190.28.114
hoogle_address = 34.98.122.184
nix_cache_ip = 35.186.227.171
$
```

**Comment by garyverhaegen-da** [comment]
I have now created a user `andreas_herrmann` and given the password to @aherrmann-da over Slack.

**Comment by garyverhaegen-da** [comment]
Updated debug template to not have preemptible instances (i.e. not kill machine every 24h)

**Comment by garyverhaegen-da** [comment]
Merging as we now need to fix hoogle deployment, and it feels cleaner not to manage separate branches. (Also master should reflect real deployments.)

**Pull request merged**